From 6df51883e64d491baa14dad71a36df4d715d9c2d Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Mon, 12 Oct 2026 10:39:56 +0200
Subject: [PATCH] Use QUrl::toDisplayString instead of QUrl::toString in error
 messages

We do not want to leak passwords.

This changes code in some backends (like the file one) that will most
probably never have a password in the url, but I decided to also change
them for consistency.

Change-Id: I2b89139582f7de26e38758c400539678571830e0
Fixes: QTBUG-151290
Reviewed-by: Kevin Ottens <kevin.ottens@enioka.com>
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 888db7669232a97fe89fcb9aed6400faf6574f75)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
Origin: upstream, https://invent.kde.org/qt/qt/qtbase/-/commit/6df51883e64d491baa14dad71a36df4d715d9c2d
Forwarded: not-needed
Last-Update: 2026-10-29
---
 src/network/access/qhttpthreaddelegate.cpp            |  6 +++---
 src/network/access/qnetworkaccesscachebackend.cpp     |  2 +-
 src/network/access/qnetworkaccessdebugpipebackend.cpp |  2 +-
 src/network/access/qnetworkaccessfilebackend.cpp      | 10 +++++-----
 src/network/access/qnetworkreplydataimpl.cpp          |  2 +-
 src/network/access/qnetworkreplyfileimpl.cpp          |  4 ++--
 src/network/access/qnetworkreplywasmimpl.cpp          |  2 +-
 7 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/src/network/access/qhttpthreaddelegate.cpp b/src/network/access/qhttpthreaddelegate.cpp
index 3b7d0e0999c..277f2549c48 100644
--- a/src/network/access/qhttpthreaddelegate.cpp
+++ b/src/network/access/qhttpthreaddelegate.cpp
@@ -82,7 +82,7 @@ static QNetworkReply::NetworkError statusCodeFromHttp(int httpStatusCode, const
             code = QNetworkReply::UnknownContentError;
         } else {
             qWarning("QNetworkAccess: got HTTP status code %d which is not expected from url: \"%s\"",
-                     httpStatusCode, qPrintable(url.toString()));
+                     httpStatusCode, qPrintable(url.toDisplayString()));
             code = QNetworkReply::ProtocolFailure;
         }
     }
@@ -507,11 +507,11 @@ static QString makeServerErrorString(int code, const QUrl &url, const QString &r
     if (!reasonPhrase.isEmpty()) {
         msg = QLatin1StringView(QT_TRANSLATE_NOOP("QNetworkReply",
                                                   "Error transferring %1 - server replied: %2"))
-                      .arg(url.toString(), reasonPhrase);
+                      .arg(url.toDisplayString(), reasonPhrase);
     } else {
         msg = QLatin1StringView(QT_TRANSLATE_NOOP("QNetworkReply",
                                                   "Error transferring %1 - server replied with status code %2"))
-                      .arg(url.toString(), QString::number(code));
+                      .arg(url.toDisplayString(), QString::number(code));
     }
     return msg;
 }
diff --git a/src/network/access/qnetworkaccesscachebackend.cpp b/src/network/access/qnetworkaccesscachebackend.cpp
index 48466e11d51..c848e5e7e3e 100644
--- a/src/network/access/qnetworkaccesscachebackend.cpp
+++ b/src/network/access/qnetworkaccesscachebackend.cpp
@@ -28,7 +28,7 @@ void QNetworkAccessCacheBackend::open()
 {
     if (operation() != QNetworkAccessManager::GetOperation || !sendCacheContents()) {
         QString msg = QCoreApplication::translate("QNetworkAccessCacheBackend", "Error opening %1")
-                                                .arg(this->url().toString());
+                                                .arg(this->url().toDisplayString());
         error(QNetworkReply::ContentNotFoundError, msg);
     } else {
         setAttribute(QNetworkRequest::SourceIsFromCacheAttribute, true);
diff --git a/src/network/access/qnetworkaccessdebugpipebackend.cpp b/src/network/access/qnetworkaccessdebugpipebackend.cpp
index e18c933b537..c53a9d6fb7e 100644
--- a/src/network/access/qnetworkaccessdebugpipebackend.cpp
+++ b/src/network/access/qnetworkaccessdebugpipebackend.cpp
@@ -153,7 +153,7 @@ void QNetworkAccessDebugPipeBackend::pushFromUpstreamToSocket()
                 if (haveWritten < 0) {
                     // write error!
                     QString msg = QCoreApplication::translate("QNetworkAccessDebugPipeBackend", "Write error writing to %1: %2")
-                                  .arg(url().toString(), socket.errorString());
+                                  .arg(url().toDisplayString(), socket.errorString());
                     error(QNetworkReply::ProtocolFailure, msg);
                     finished();
                     return;
diff --git a/src/network/access/qnetworkaccessfilebackend.cpp b/src/network/access/qnetworkaccessfilebackend.cpp
index 049b1d4d392..96a2ab4f98a 100644
--- a/src/network/access/qnetworkaccessfilebackend.cpp
+++ b/src/network/access/qnetworkaccessfilebackend.cpp
@@ -86,7 +86,7 @@ void QNetworkAccessFileBackend::open()
     if (!url.host().isEmpty()) {
         // we handle only local files
         error(QNetworkReply::ProtocolInvalidOperationError,
-              QCoreApplication::translate("QNetworkAccessFileBackend", "Request for opening non-local file %1").arg(url.toString()));
+              QCoreApplication::translate("QNetworkAccessFileBackend", "Request for opening non-local file %1").arg(url.toDisplayString()));
         finished();
         return;
     }
@@ -140,7 +140,7 @@ void QNetworkAccessFileBackend::open()
     // could we open the file?
     if (!opened) {
         QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Error opening %1: %2")
-                                                .arg(this->url().toString(), file.errorString());
+                                                .arg(this->url().toDisplayString(), file.errorString());
 
         // why couldn't we open the file?
         // if we're opening for reading, either it doesn't exist, or it's access denied
@@ -179,7 +179,7 @@ void QNetworkAccessFileBackend::uploadReadyReadSlot()
             if (haveWritten < 0) {
                 // write error!
                 QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Write error writing to %1: %2")
-                              .arg(url().toString(), file.errorString());
+                              .arg(url().toDisplayString(), file.errorString());
                 error(QNetworkReply::ProtocolFailure, msg);
 
                 finished();
@@ -212,7 +212,7 @@ bool QNetworkAccessFileBackend::loadFileInfo()
 
     if (fi.isDir()) {
         error(QNetworkReply::ContentOperationNotPermittedError,
-              QCoreApplication::translate("QNetworkAccessFileBackend", "Cannot open %1: Path is a directory").arg(url().toString()));
+              QCoreApplication::translate("QNetworkAccessFileBackend", "Cannot open %1: Path is a directory").arg(url().toDisplayString()));
         finished();
         return false;
     }
@@ -236,7 +236,7 @@ qint64 QNetworkAccessFileBackend::read(char *data, qint64 maxlen)
         // EOF or error
         if (file.error() != QFile::NoError) {
             QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Read error reading from %1: %2")
-                            .arg(url().toString(), file.errorString());
+                            .arg(url().toDisplayString(), file.errorString());
             error(QNetworkReply::ProtocolFailure, msg);
 
             finished();
diff --git a/src/network/access/qnetworkreplydataimpl.cpp b/src/network/access/qnetworkreplydataimpl.cpp
index e9cb01f09fc..e1c1ced014d 100644
--- a/src/network/access/qnetworkreplydataimpl.cpp
+++ b/src/network/access/qnetworkreplydataimpl.cpp
@@ -54,7 +54,7 @@ QNetworkReplyDataImpl::QNetworkReplyDataImpl(QObject *parent, const QNetworkRequ
     } else {
         // something wrong with this URI
         const QString msg = QCoreApplication::translate("QNetworkAccessDataBackend",
-                                                        "Invalid URI: %1").arg(url.toString());
+                                                        "Invalid URI: %1").arg(url.toDisplayString());
         setError(QNetworkReply::ProtocolFailure, msg);
         QMetaObject::invokeMethod(this, "errorOccurred", Qt::QueuedConnection,
                                   Q_ARG(QNetworkReply::NetworkError, QNetworkReply::ProtocolFailure));
diff --git a/src/network/access/qnetworkreplyfileimpl.cpp b/src/network/access/qnetworkreplyfileimpl.cpp
index ed75875ee3a..789e12e8f3e 100644
--- a/src/network/access/qnetworkreplyfileimpl.cpp
+++ b/src/network/access/qnetworkreplyfileimpl.cpp
@@ -52,7 +52,7 @@ QNetworkReplyFileImpl::QNetworkReplyFileImpl(QNetworkAccessManager *manager, con
     // do not allow UNC paths on Unix
     if (!url.host().isEmpty()) {
         // we handle only local files
-        QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Request for opening non-local file %1").arg(url.toString());
+        QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Request for opening non-local file %1").arg(url.toDisplayString());
         setError(QNetworkReply::ProtocolInvalidOperationError, msg);
         setFinished(true); // We're finished, will emit finished() after ctor is done.
         QMetaObject::invokeMethod(this, "errorOccurred", Qt::QueuedConnection,
@@ -98,7 +98,7 @@ QNetworkReplyFileImpl::QNetworkReplyFileImpl(QNetworkAccessManager *manager, con
 
         QFileInfo fi(fileName);
         if (fi.isDir()) {
-            QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Cannot open %1: Path is a directory").arg(url.toString());
+            QString msg = QCoreApplication::translate("QNetworkAccessFileBackend", "Cannot open %1: Path is a directory").arg(url.toDisplayString());
             setError(QNetworkReply::ContentOperationNotPermittedError, msg);
             QMetaObject::invokeMethod(this, "errorOccurred", Qt::QueuedConnection,
                 Q_ARG(QNetworkReply::NetworkError, QNetworkReply::ContentOperationNotPermittedError));
diff --git a/src/network/access/qnetworkreplywasmimpl.cpp b/src/network/access/qnetworkreplywasmimpl.cpp
index 55b3a0312e5..984f75ae427 100644
--- a/src/network/access/qnetworkreplywasmimpl.cpp
+++ b/src/network/access/qnetworkreplywasmimpl.cpp
@@ -683,7 +683,7 @@ QNetworkReply::NetworkError QNetworkReplyWasmImplPrivate::statusCodeFromHttp(int
             code = QNetworkReply::UnknownContentError;
         } else {
             qWarning("QNetworkAccess: got HTTP status code %d which is not expected from url: \"%s\"",
-                     httpStatusCode, qPrintable(url.toString()));
+                     httpStatusCode, qPrintable(url.toDisplayString()));
             code = QNetworkReply::ProtocolFailure;
         }
     };
